File: python/04-lista/main083.py
'''[DESAFIO] Crie uma lógica que preencha um vetor de 20 posições com números 
aleatórios (entre 0 e 99) gerados pelo computador. Logo em seguida, mostre os 
números gerados e depois coloque o vetor em ordem crescente, mostrando no final 
os valores ordenados.

Data 04/11/2025 -> 42ª Dia
'''

import random

lista = []

for i in range(20):
    aleatorio = random.randint(0,99)
    lista.append(aleatorio)
ordenado = sorted(lista)
print(f'Forma não ordenada: {lista}')
print(f'Forma ordenada: {ordenado}')

File: c/04-vetores/main083.c
/* [DESAFIO] Crie uma lógica que preencha um vetor de 20 posições com números 
aleatórios (entre 0 e 99) gerados pelo computador. Logo em seguida, mostre os 
números gerados e depois coloque o vetor em ordem crescente, mostrando no final 
os valores ordenados.

Data 04/11/2025 -> 42ª Dia
*/

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(){

    int vetor[20];


    srand(time(NULL));
    for(int i = 0; i < 20; i++){

        vetor[i] = rand() % 100; 

    }
    printf("Forma nao ordenada: ");
    for(int i = 0; i < 20; i++){

        printf("%d ",vetor[i]);
    }
    for(int i = 0; i < 20 - 1; i++){
        for(int j = 0; j < 20-i-1; j++){
            if(vetor[j] > vetor[j+1]){
                int temp = vetor[j];
                vetor[j] = vetor[j+1];
                vetor[j+1] = temp;
            }
        }
    }
    printf("\nForma ordenada: ");
    for(int i = 0; i < 20; i++){

        printf("%d ",vetor[i]);
    }
    return 0;
}